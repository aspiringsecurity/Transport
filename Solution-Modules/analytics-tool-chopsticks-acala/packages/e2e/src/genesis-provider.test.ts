import { describe, expect, it } from 'vitest'

import { api, chain, dev, env, expectJson, setupApi, testingPairs } from './helper'

setupApi(env.mandalaGenesis)

describe('genesis provider works', () => {
  it('build blocks', async () => {
    expect(await dev.newBlock()).toMatchInlineSnapshot(
      '"0xcacc274d53b81070033e1b14b4162917be2754ecffaa31943cff06c39d2a8720"'
    )
    const block = await chain.getBlock('0xcacc274d53b81070033e1b14b4162917be2754ecffaa31943cff06c39d2a8720')
    expect(block).toBeTruthy
    expect(block?.number).toBe(1)
  })

  it('handles tx', async () => {
    await dev.newBlock()

    const { alice, bob } = testingPairs()

    await dev.setStorage({
      System: {
        Account: [[[alice.address], { data: { free: 1000 * 1e12 } }]],
      },
    })

    expectJson(await api.query.system.account(alice.address)).toMatchSnapshot()

    await api.tx.currencies.transferNativeCurrency(bob.address, 100 * 1e12).signAndSend(alice)

    await dev.newBlock()

    expectJson(await api.query.system.account(alice.address)).toMatchSnapshot()
    expectJson(await api.query.system.account(bob.address)).toMatchSnapshot()
  })
})
