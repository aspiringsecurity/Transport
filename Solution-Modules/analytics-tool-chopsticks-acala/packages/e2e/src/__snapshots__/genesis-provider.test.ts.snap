// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`genesis provider works > handles tx 1`] = `
{
  "consumers": 0,
  "data": {
    "feeFrozen": 0,
    "free": 1000000000000000,
    "miscFrozen": 0,
    "reserved": 0,
  },
  "nonce": 0,
  "providers": 0,
  "sufficients": 0,
}
`;

exports[`genesis provider works > handles tx 2`] = `
{
  "consumers": 0,
  "data": {
    "feeFrozen": 0,
    "free": 899981152759869,
    "miscFrozen": 0,
    "reserved": 0,
  },
  "nonce": 1,
  "providers": 0,
  "sufficients": 0,
}
`;

exports[`genesis provider works > handles tx 3`] = `
{
  "consumers": 0,
  "data": {
    "feeFrozen": 0,
    "free": 100000000000000,
    "miscFrozen": 0,
    "reserved": 0,
  },
  "nonce": 0,
  "providers": 1,
  "sufficients": 0,
}
`;
